free42: 3.3.8 -> 3.3.10, plus42: init at 1.3.12

elfenermarcell opened the pull request:
This changes free42 to use the release source tarballs, as it means 1) we only download the part needed for the linux build, 2) it doesn't contain the HP logo (potential legal issue, I think).
It also makes it so free42 actually uses alsa, it wasn't actually built with it despite having the dependency.

And this also adds plus42, an enhanced version of free42 by the same author. The build is basically the same.
## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

elfenermarcell commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 449165`
Commit: `c3064473e75193edaad2957332af1270e8d68be7`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>free42</li>
    <li>plus42</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>free42</li>
    <li>plus42</li>
  </ul>
</details>

elfenermarcell commented:
Nevermind, the audio on wayland issue is a bug in the programs, opened issue upstream https://codeberg.org/thomasokken/free42/issues/84

qweered reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.

elfenermarcell committed:
free42: 3.3.8 -> 3.3.10

This changes the source as recommended on the program's website because
1) it means we only download the part needed for the linux build
2) it doesn't contain the HP logo
and makes it so the alsa dependency actually gets used

elfenermarcell committed:
plus42: init at 1.3.12

This is mostly a copy of free42's package.nix, just with the name changed.

elfenermarcell commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 449165`
Commit: `be9feade2e9663a0ae02a6d20d36048daa84f67e`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>free42</li>
    <li>plus42</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>free42</li>
    <li>plus42</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/by-name/fr/free42/package.nix:
```diff
@@ -1,24 +1,21 @@
+  version = "3.3.10";
+
+  src = fetchurl {
+    url = "https://thomasokken.com/free42/upstream/free42-nologo-${version}.tgz";
```
Why do we no longer fetch the primary source?

elfenermarcell replied:
On the projects' website (https://thomasokken.com/free42/) it links to this source, with the text "Package for GTK only, no HP logo, for building Linux packages".

Other distros also use this source, presumably because distributing the HP logo may have legal issues. (And the Debian package is literally called `free42-nologo`)

SuperSandro2000 commented on pkgs/by-name/pl/plus42/package.nix:
```diff
@@ -0,0 +1,119 @@
+  version = "1.3.12";
+
+  src = fetchurl {
+    url = "https://thomasokken.com/plus42/upstream/plus42-upstream-${version}.tgz";
```
same here, why not fetch directly source?
